framework.py

```diff
@@ -122,21 +124,15 @@ def _remove_old_cached_data(self, grace_period, *cache_locations):
         Remove old data files cached by this bundle in the given cache locations.
 
         :param int grace_period: Time period, in days, a file without
-                                     modification should be kept around.
+                                 modification should be kept around.
         :param cache_locations: A list of cache locations to clean up.
         :raises: ValueError if the grace period is smaller than one day.
         """
         if grace_period < 1:
             raise ValueError(
                 "Invalid grace period value %d, it must be a least 1" % grace_period
             )
-        delta = datetime.timedelta(days=grace_period)
```

Comment:
This unintuitive piece of code caused an error with Python 3.7 due to the use of a true division. I've simplified the code since it introduced errors in Python 3 anyway. Both this code and it's test now use the much simpler assumption that `24 * 3600` is a full day. We're trying to roughly count days, not be super precise to the millisecond about it anyway.

Reply:
But what if the number of days in the year or hours in the day changes??  😆 (This is way easier to read, makes sense.)